src/dfx/src/commands/replica.rs

```diff
@@ -178,8 +188,31 @@ pub fn exec(
     system.block_on(async move {
         let shutdown_controller = start_shutdown_controller(env)?;
         if emulator {
+            let effective_config = CachedConfig::emulator();
+            if !force && previous_config_path.exists() {
+                let previous_config = load_json_file(&previous_config_path).context(
+                    "Failed to read replica configuration. Run `dfx start` with `--clean`.",
+                )?;
+                if effective_config != previous_config {
+                    bail!("The network configuration was changed. Run `dfx start` with `--clean`.")
+                }
+            }
+            save_json_file(&previous_config_path, &effective_config)
+                .context("Failed to write replica configuration")?;
             start_emulator_actor(env, shutdown_controller, emulator_port_path)?;
         } else {
+            let effective_config = CachedConfig::replica(&replica_config);
+            if !force && previous_config_path.exists() {
+                let previous_config = load_json_file(&previous_config_path).context(
+                    "Failed to read replica configuration. Run `dfx start` with `--clean`.",
+                )?;
+                if effective_config != previous_config {
+                    bail!("The network configuration was changed. Run `dfx start` with `--clean`.")
+                }
+            }
+            save_json_file(&previous_config_path, &effective_config)
+                .context("Failed to write replica configuration")?;
+
```

Comment:
I wanted it to look basically the same as the code in start.rs, which *does* need to stay that way, so both can easily be updated at the same time. But if you say so.

Reply:
Understood.  My intent was to follow up with a change to start.rs to move that configuration out of the block, since I didn't want to ask you to do that here as it's adjacent work.